Algorithm.CSharp/CustomDataTypeHistoryAlgorithm.cs

```diff
@@ -0,0 +1,170 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Algorithm asserting that
+    /// </summary>
+    public class CustomDataTypeHistoryAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _symbol;
+
+        public override void Initialize()
+        {
+            SetStartDate(2017, 8, 20);
+            SetEndDate(2017, 8, 20);
+
+            _symbol = AddData<CustomDataType>("CustomDataType", Resolution.Hour).Symbol;
+
+            var history = History<CustomDataType>(_symbol, 48, Resolution.Hour).ToList();
+
+            Log($"History count: {history.Count}");
+
+            if (history.Count == 0)
+            {
+                throw new Exception("History request returned no data");
+            }
+
+            var history2 = History<CustomDataType>(new[] { _symbol }, 48, Resolution.Hour).ToList();
+
+            if (history2.Count != history.Count)
+            {
+                throw new Exception("History requests returned different data");
+            }
+
+        }
+
+        public class CustomDataType : DynamicData
```

Comment:
nit - could this class be private? to avoid exposing it to others if not required 👍